### PR TITLE
[e2e] Setup MV3 specific test folder and add 1 simple testcase introducing Service Worker re-start

### LINKED ISSUE #16610: [e2e] Setup MV3 specific tests introducing Service Worker re-start

### PR DESCRIPTION
**Pull request by seaona**
## Explanation
There are some specific behaviors for the MV3 build that need to be covered by our e2e tests, related to stopping and re-starting service worker between user actions.

For that, this PR sets the initial base work and adds one testcase, so this can be continued by adding more testcases in the future. The work consists on:
- Add mv3 flag for running e2e tests
- Include mv3 flag on circle ci job for `e2e:chrome:mv3`
- Add specific folder for MV3 tests
- Add simple testcase to check dapp interaction after a service worker re-start. 

At the moment, this will fail for 2 reasons:
  - Keep the user logged PR is still not merged
  - Popup issue after terminating service worker. See here: https://github.com/MetaMask/metamask-extension/issues/16554

Fixes #16610

### After

image

## Manual Testing Steps
Check [circle ci ](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/31189/workflows/abb51914-1393-4eb7-b0cd-aa14c533e1d3/jobs/814648/parallel-runs/6?filterBy=ALL)and see that the new test is run on the `e2e:chrome:mv3` job.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by Gudahtt — DISMISSED** [review]
Great work! Left a couple suggestions, nothing blocking though.

File: test/e2e/run-all.js [hunk 1]
```diff
@@ -56,7 +60,14 @@ async function main() {
 
   let testPaths = await getTestPathsForTestDir(testDir);
 
-  if (!snaps) {
+  if (mv3) {
+    testPaths = [
+      ...testPaths,
+      ...(await getTestPathsForTestDir(path.join(__dirname, 'swaps'))),
```
[the remark this answers is not in the document]
Comment (seaona) [reply test/e2e/run-all.js]: thank you for your suggestion @Gudahtt . This makes more sense

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: test/e2e/mv3/dapp-interactions.spec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          windowHandles,
+        );
+
+        const notification = await driver.isElementPresent({
+          text: 'Allow this site to add a network?',
+          tag: 'h3',
+        });
+
+        assert.ok(notification, 'Dapp action does not appear in Metamask');
+      },
```
Comment (jpuri) [inline test/e2e/mv3/dapp-interactions.spec.js:63]: This message is un-intuitive: **Dapp action does not appear in Metamask**

**Review by jpuri — APPROVED** [review]

**Pull request merged**